api/src/wfl/service/slack.clj

```diff
@@ -77,12 +84,17 @@
 
 ;; FIXME: add permission checks for slack-channel-watchers
 ;;
-(defn notify-channels
-  "Notify `watchers` of an `exception` in the workload with `uuid`."
-  [channels uuid exception]
-  (let [message        (format "Workload %s update threw %s" uuid (.getMessage exception))]
+(defn notify-watchers
+  "Send `message` associated with workload `uuid` to Slack `watchers`."
+  [{:keys [watchers uuid project labels] :as _workload} message]
+  (let [channels (filter slack-channel-watcher? watchers)
+        project  (or project (util/label-value labels "project"))
```

Comment:
Annoyingly, loading a staged workload doesn't expose the project directly, we need to pull it from the labels list.

Reply:
Better name though.